### PR TITLE
Add eWASM schedule

### PR DESCRIPTION
**Pull request by chfast**
It's Byzantium + unlimited code size

### CODE REVIEW

**Comment by axic** [comment]
If we could I would also remove the ~~dumb~~ weird idea of not charging for 0 bytes in contract code in `eWASMSchedule` :)

**Comment by axic** [comment]
So apparently it can be done:
```
 	unsigned txDataZeroGas = 4;
 	unsigned txDataNonZeroGas = 68;
```

I'd suggest going with `txDataZeroGas = txDataNonZeroGas` in `EWASMSchedule`.

File: libethcore/EVMSchedule.h [hunk 1]
```diff
@@ -116,6 +116,13 @@ static const EVMSchedule ByzantiumSchedule = []
 	return schedule;
 }();
 
+static const EVMSchedule EWASMSchedule = []
+{
+    EVMSchedule schedule = ByzantiumSchedule;
+    schedule.maxCodeSize = std::numeric_limits<unsigned>::max();
```
Comment (winsvega) [inline libethcore/EVMSchedule.h]: which EIP is this?

Comment (chfast) [reply libethcore/EVMSchedule.h]: The Lisbon spring EIP.

File: libethereum/ChainParams.cpp [hunk 2]
```diff
@@ -70,6 +70,7 @@ string const c_daoHardforkBlock = "daoHardforkBlock";
 string const c_EIP150ForkBlock = "EIP150ForkBlock";
 string const c_EIP158ForkBlock = "EIP158ForkBlock";
 string const c_byzantiumForkBlock = "byzantiumForkBlock";
+string const c_eWASMForkBlock = "eWASMForkBlock";
```
Comment (winsvega) [inline libethereum/ChainParams.cpp]: with another fork I think it would be reasonable to make markers in tests. like 
execute this test on  Frontier, Homestead  but not Metropolis. 
and so on. this way we will reduce the amount of tests.

File: libethcore/EVMSchedule.h (hunk 1 above)
Comment (winsvega) [reply libethcore/EVMSchedule.h]: link?

File: libethereum/ChainParams.cpp (hunk 2 above)
Comment (chfast) [reply libethereum/ChainParams.cpp]: I don't fully understand, but some improvements in configuration would be nice. I have some ideas now. We should also discuss this with geth team. They should have more experience.

Also, we can use YAML now for better readability.

**Comment by chfast** [comment]
Looks like every test fails now :D Nice.

**Comment by winsvega** [comment]
what is the default forkblock number value?

**Comment by chfast** [comment]
I have not set any, so probably 0. Where do set the default value?

**Comment by winsvega** [comment]
the value is defined in genesis config files. 
https://github.com/ethereum/cpp-ethereum/tree/develop/libethashseal/genesis

**Comment by axic** [comment]
Just copy constanintople:
```
		"constantinopleForkBlock": "0xffffffffffffffffff",
```

**Comment by chfast** [comment]
Pushed one more commit. It sets default value for fork block numbers in infinity (instead of 0), so we don't have to put these magic `0xffffff...` for every new fork in every config file.

File: test/unittests/libethashseal/EthashTest.cpp [hunk 3]
```diff
@@ -48,12 +52,13 @@ BOOST_AUTO_TEST_CASE(calculateDifficultyByzantiumWithoutUncles)
 	header.setNumber(0x2001);
 	header.setTimestamp(130);
 
-	BOOST_REQUIRE_EQUAL(ethash.calculateDifficulty(header, parentHeader), 999024);
+	BOOST_CHECK_EQUAL(ethash.calculateDifficulty(header, parentHeader), 999024);
```
Comment (gumb0) [inline test/unittests/libethashseal/EthashTest.cpp]: Why did you change this?

Comment (chfast) [reply test/unittests/libethashseal/EthashTest.cpp]: Just pedantic mode. I prefer weaker check if available.
